lib/Sema/CodeSynthesis.cpp

```diff
@@ -763,17 +767,25 @@ void swift::addTrivialAccessorsToStorage(AbstractStorageDecl *storage,
 
   // Synthesize and type-check the body of the getter.
   synthesizeTrivialGetter(getter, storage, TC);
-  TC.typeCheckDecl(getter, true);
-  TC.typeCheckDecl(getter, false);
+  if (storage->getDeclContext()->isModuleScopeContext()) {
```

Comment:
Without that check, various resilience tests fail when emitting global getters (compiler crashes). I imagine this is due to unconditional emission of those getters, w/o being driven by any other references.

Seems to work when I tighten the condition to "only when in resilient mode", so I'll try that (and add a comment to that extent).